src/components/atoms/icon.tsx

```diff
@@ -6,9 +6,22 @@ export interface IconProps {
   name?: IconName;
   color?: string;
   size: number;
+  url?: string;
 }
 
-const Icon = ({ name = undefined, color = undefined, size }: IconProps) =>
+const BaseIcon = ({ name = undefined, color = undefined, size }: Omit<IconProps, 'url'>) =>
   name ? <IconComponent icon={name} width={size} height={size} style={{ color }} /> : null;
 
+const Icon = ({ url = undefined, ...baseIconProps }: IconProps) => {
```

Comment:
Wydaje mi się, że ikonka jako taka nie powinna móc być linkiem. Utrudnia to customizację poprzez props'y. Nie możemy na przykład ustawić innego targetu albo `rel`. Może lepiej bezpośredion owrapować sobie ikonkę w `a` tam, gdzie jest to potrzebne

Reply:
W takim razie można by dodać handling dla linków do komponentu tag, bo to jest główny case, w którym występuje ikonka + link. Co sądzisz o czymś takim? 🤔 

```astro
---
import type { IconName } from '@/types/icon';

import Icon from './icon';

export interface Props {
  icon?: {
    name: IconName;
    color?: string;
  };
  anchorProps?: astroHTML.JSX.AnchorHTMLAttributes;
}

const {
  props: { icon },
  anchorProps,
} = Astro;

const Wrapper = anchorProps ? "a" : "div"
---

<Wrapper class="flex items-center w-fit h-6 px-2.5 rounded text-sm font-medium text-gray-700 bg-gray-100 gap-x-1.5">
  {icon && <Icon client:load name={icon.name} color={icon.color} size={16} />}
  <slot />
</Wrapper>

```